# sc_rna_seq
A repository containing usable code for processing and analysis of single-cell RNAseq data. 

## Pre-requisites
Usage of this code expects familiarity with the following:
* Linux/Unix command-line system
* R programming language
* computational single-cell RNA sequencing analysis

As of now, the constituent scripts (`src/quality_control.R`) expects the user to have completed pre-processing of the raw FASTQ files using `cellranger` and thus, to have generated files corresponding to the:

* filtered cell-by-gene count data
* raw count data
* cluster assignments of each cell generated by `cellranger`

## Installation
Prior to using the code in this repository, the user is expected to have the following installed in their computing environment:

* R programming language 
* the following R packages
    * `jsonlite`
    * `ggplot2`
    * `tidyverse`
    * `Matrix`
    * `Seurat`
    * `SeuratObject`
    * `scales`
    * `harmony`
    * `cowplot`
    * `purrr`
    * `plyr`
    * `patchwork`
    * `SoupX`
    * `DoubletFinder`

## Usage

* open the configuration file `cfg/quality_control.json`
* specify all the required parameters 
* save the configuration file
* run the QC script
```bash
Rscript src/quality_control.R
```

## Authors
Al Latif